### ISSUE TITLE
Running in multithreaded environment 

### ISSUE THREAD

**Issue by kayoubi** [rinoto/spring-request-cache#1]
Won't spawning new thread in the request cause some issues ? e.g. each thread will have it's own cache and evicting in one won't affect the others ? same applies when starting new transaction with something like `REQUIRED_NEW` ..

**Comment by rinoto** [comment]
> each thread will have it's own cache and evicting in one won't affect the others

Yes, that's the point: the cache will only be available for the request, i.e. for the thread we are in

> same applies when starting new transaction with something like REQUIRED_NEW

You are right here. If the request creates a new thread, the cache will not be available in the created thread

**Comment by vadivelselvaraj** [comment]
I think this cache would break when the thread gets reused for another request(like in a web server that does ThreadPool). 

Also, I think caching at the request level can be done easily using @Scope annotation with value 'request'. An instance would be this --  https://dzone.com/articles/spring-request-level

**Comment by rinoto** [comment]
> I think this cache would break when the thread gets reused for another request(like in a web server that does ThreadPool).

Right. That's why I stated in the docu what you need to do against it.

> Also, I think caching at the request level can be done easily using @Scope annotation with value 'request'. An instance would be this -- https://dzone.com/articles/spring-request-level

It seems that we have a different concept of "easily" ;) 

Thanks for the comments!
